cloudinit/distros/tests/test_init.py

```diff
@@ -154,3 +164,32 @@ def test_valid_substitution(self,
         print(patterns)
         print(expected)
         assert {'primary': expected} == ret
+
+
+class TestNetworking:
+    """Test the Distro.networking attribute."""
+
+    def test_missing_networking(self, concrete_distro_cls):
+        """Emulate a pre-upgrade `Distro` and confirm .networking exists.
+
+        When a distro instance that was created before the networking refactor
+        was started is unpickled, it will not have the .networking instance
+        attribute.  This test ensures that we handle this case.
+        """
+        distro = concrete_distro_cls("", {}, mock.Mock())
+        for attr in ["networking", "_networking"]:
+            try:
+                delattr(distro, attr)
+            except AttributeError:
+                # This either means the attribute is already absent, or it is
+                # part of the class definition and therefore (a) cannot be
+                # removed, and (b) will be present on unpickled objects.  In
+                # both cases, we do not need to remove the attribute.
+                pass
+
+        assert distro.networking is not None
+
+    def test_same_networking_instance_returned(self, concrete_distro_cls):
+        """.networking should always return the same Networking instance."""
+        distro = concrete_distro_cls("", {}, mock.Mock())
+        assert distro.networking is distro.networking
```

Comment:
Shouldn't we also assert that distro.networking is not None before asserting they're equal?